File: 5 - Linguagens de Programação/Aulas/Aula08 - Efeitos Inesperados em LPs/side_effect.c
#include <stdio.h>

int b;

int func(int a) {
    b = 20;
    return a * a;
}

int main() {
    int a = 2;
    b = 10;
    int c = 2;
    printf("%d\n", (b + func(a)) / (func(a) - c));
    int temp = func(a);
    printf("%d\n", (temp + b) / (temp - c));
}

File: 5 - Linguagens de Programação/Aulas/Aula08 - Efeitos Inesperados em LPs/side_effect.py
# transparência referencial
# b = 0

# def func(a):
#     global b
#     b = 20
#     return a * a

# a = 2
# b = 10
# c = 2
# print((func(a) + b) // (func(a) - c))
# temp = func(a)
# print((temp + b) // (temp - c))

# listas de atribuição e troca
# possíveis em Perl, Ruby, Python, Lua ...

x, y = 1, 6
p = [7, 8]
a, b = p
a, b = b, a
print(a, b)